documentation/modules/deploying/con-switching-api-to-kafka-connector.adoc

```diff
@@ -0,0 +1,18 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deploy-kafka-connect-with-plugins.adoc
+
+[id='con-switching-api-to-kafka-connector-{context}']
+= Switching from using the Kafka Connect API to using KafkaConnector custom resources
+
+[role="_abstract"]
+You can switch from using the Kafka Connect API to using `KafkaConnector` custom resources to manage your connectors.
+To make the switch, do the following in the order shown:
+
+. Deploy `KafkaConnector` resources with the configuration to create your connector instances.
+. Enable `KafkaConnector` resources in your Kafka Connect configuration by setting the `strimzi.io/use-connector-resources` annotation to `true`.
+
+WARNING: If you enable `KafkaConnector` resources before creating them, you delete all connectors.
+
+To switch from using `KafkaConnector` resources to using the Kafka Connect API, first remove the annotation that enables the `KafkaConnector` resources from your Kafka Connect configuration.
```

Comment:
Are we missing something here that says when you can safely delete your KafkaConnector resources? How does the user know the operator has picked up the switch across?

Reply:
Good question. @scholzj --  do we want to say something about verifying the switch (deleting kafkaconnectors safely)?

Reply:
Well, basically as with every change: you would compare the `.metadata.generation` after disabling the connector operator with the `.status.observedGeneration`. The Connect cluster should be `Ready` and these two should be equal. Do we have some general chapter for this? If not, it might make sense to have it and link there. Something like _How do you know your changes have been applied_. But we already might have it covered somewhere.

Reply:
Thanks @scholzj -- we do have a section of checking status, so I've linked as follows:

> When making the switch, link:{BookURLConfiguring}#con-custom-resources-status-str[check the status of the `KafkaConnect` resource]. The value of `metadata.generation` (the current version of the deployment) must match `status.observedGeneration` (the latest reconciliation of the resource). When the Kafka Connect cluster is `Ready`, you can delete the `KafkaConnector` resources.